packages/discount-code-exporter/test/main.spec.js

```diff
@@ -57,70 +57,146 @@ describe('DiscountCodeExport', () => {
     })
   })
 
-  describe('::run', () => {
-    test('should fetch discount codes and output csv to stream', done => {
-      codeExport.config.exportFormat = 'csv'
-      const sampleCode = {
-        code: 'discount-code',
-        name: { en: 'some-discount-name' },
-        cartDiscounts: [{ id: 'cart-discount-1' }, { id: 'cart-discount-2' }],
-      }
-      codeExport._fetchCodes = jest.fn().mockImplementation(csvStream => {
-        csvStream.write(sampleCode)
-        return Promise.resolve()
-      })
-      const outputStream = streamtest.v2.toText((error, result) => {
-        const expectedResult = stripIndent`
-          code,name.en,cartDiscounts,groups
-          discount-code,some-discount-name,cart-discount-1;cart-discount-2,
-          `
-        expect(result).toEqual(expectedResult)
-        done()
+  describe('::setupHeaders', () => {
+    describe('with passed headers', () => {
+      test('should return return passed headers', () => {
+        expect(DiscountCodeExport.setupHeaders(['foo', 'bar'], 'de')).toEqual([
+          'foo',
+          'bar',
+        ])
       })
-      codeExport.run(outputStream)
     })
 
-    test('should fetch codes and output json to stream by default', done => {
-      const sampleCode = {
-        code: 'discount-code',
-        name: { en: 'some-discount-name' },
-        cartDiscounts: [{ id: 'cart-discount-1' }, { id: 'cart-discount-2' }],
-      }
-      codeExport._fetchCodes = jest.fn().mockImplementation(jsonStream => {
-        jsonStream.write(sampleCode)
-        return Promise.resolve()
+    describe('without passed headers', () => {
+      describe('when empty array', () => {
+        test('should return return passed headers', () => {
+          expect(DiscountCodeExport.setupHeaders([], 'de')).toMatchSnapshot()
```

Comment:
Nitpick: but for this I would not use a snapshot. I sense it'd be easier to have `expect(DiscountCodeExport.setupHeaders([], 'de')).toEqual([])` as then the expected value is co-located. It's not like it will change or contains values and the nice CLI diffing of snapshotting it provides any benefit.

Reply:
I'm actually not expecting an empty array to be returned in this case, but yeah. snapshotting doesn't help here